drivers/sensor/lsm6dsl/lsm6dsl.c

```diff
@@ -0,0 +1,447 @@
+	}
+
+	return 0;
+}
+
+static int lsm6dsl_gyro_set_fs_raw(struct device *dev, u8_t fs)
+{
+	struct lsm6dsl_data *data = dev->driver_data;
+	const struct lsm6dsl_config *config = dev->config->config_info;
+
+	if (fs == GYRO_FULLSCALE_125) {
+		if (i2c_reg_update_byte(data->i2c_master,
+					config->i2c_slave_addr,
+					LSM6DSL_REG_CTRL2_G,
+					LSM6DSL_MASK_CTRL2_FS125,
+					1 << LSM6DSL_SHIFT_CTRL2_FS125) < 0) {
+			return -EIO;
+		}
+	} else {
+		if (i2c_reg_update_byte(data->i2c_master,
+					config->i2c_slave_addr,
+					LSM6DSL_REG_CTRL2_G,
+					LSM6DSL_MASK_CTRL2_G_FS_G,
+					fs << LSM6DSL_SHIFT_CTRL2_G_FS_G) < 0) {
+			return -EIO;
+		}
+	}
+
+	return 0;
+}
+
+static int lsm6dsl_gyro_set_odr_raw(struct device *dev, u8_t odr)
+{
+	struct lsm6dsl_data *data = dev->driver_data;
+	const struct lsm6dsl_config *config = dev->config->config_info;
+
+	if (i2c_reg_update_byte(data->i2c_master, config->i2c_slave_addr,
+				LSM6DSL_REG_CTRL2_G,
+				LSM6DSL_MASK_CTRL2_G_ODR_G,
+				odr << LSM6DSL_SHIFT_CTRL2_G_ODR_G) < 0) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int lsm6dsl_sample_fetch_accel(struct device *dev)
+{
+	struct lsm6dsl_data *data = dev->driver_data;
+	const struct lsm6dsl_config *config = dev->config->config_info;
+	u8_t buf[6];
+
+
+	if (i2c_burst_read(data->i2c_master, config->i2c_slave_addr,
+			   LSM6DSL_REG_OUTX_L_XL, buf, sizeof(buf)) < 0) {
+		SYS_LOG_DBG("failed to read sample");
+		return -EIO;
+	}
+
+	data->accel_sample_x = (s16_t)((u16_t)(buf[0]) |
+				((u16_t)(buf[1]) << 8));
+	data->accel_sample_y = (s16_t)((u16_t)(buf[2]) |
+				((u16_t)(buf[3]) << 8));
+	data->accel_sample_z = (s16_t)((u16_t)(buf[4]) |
+				((u16_t)(buf[5]) << 8));
+
+	return 0;
+}
+
+static int lsm6dsl_sample_fetch_gyro(struct device *dev)
+{
+	struct lsm6dsl_data *data = dev->driver_data;
+	const struct lsm6dsl_config *config = dev->config->config_info;
+	u8_t buf[6];
+
+	if (i2c_burst_read(data->i2c_master, config->i2c_slave_addr,
+			   LSM6DSL_REG_OUTX_L_G, buf, sizeof(buf)) < 0) {
+		SYS_LOG_DBG("failed to read sample");
+		return -EIO;
+	}
+
+	data->gyro_sample_x = (s16_t)((u16_t)(buf[0]) |
+				((u16_t)(buf[1]) << 8));
+	data->gyro_sample_y = (s16_t)((u16_t)(buf[2]) |
+				((u16_t)(buf[3]) << 8));
+	data->gyro_sample_z = (s16_t)((u16_t)(buf[4]) |
+				((u16_t)(buf[5]) << 8));
+
+	return 0;
+}
+
+#if defined(CONFIG_LSM6DSL_ENABLE_TEMP)
+static int lsm6dsl_sample_fetch_temp(struct device *dev)
+{
+	struct lsm6dsl_data *data = dev->driver_data;
+	const struct lsm6dsl_config *config = dev->config->config_info;
+	u8_t buf[2];
+
+	if (i2c_burst_read(data->i2c_master, config->i2c_slave_addr,
+			   LSM6DSL_REG_OUT_TEMP_L, buf, sizeof(buf)) < 0) {
+		SYS_LOG_DBG("failed to read sample");
+		return -EIO;
+	}
+
+	data->temp_sample = (s16_t)((u16_t)(buf[0]) |
+				((u16_t)(buf[1]) << 8));
+
+	return 0;
+}
+#endif
+
+static int lsm6dsl_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL ||
```

Comment:
Should also add SENSOR_CHAN_ACCEL_XYZ and SENSOR_CHANN_TEMP. Or just remove the assert altogether, since you have the switch default return -ENOTSUP.